less_users/views.py

```diff
@@ -40,104 +48,231 @@ class ProfileView(LoginRequiredMixin, View):
-        this.delete()
-        messages.warning(request, f'You have deleted {this.challenge.name} from your challenges')
-    return redirect('my_challenges')
+        UserChallenge.objects.create(user=request.user, challenge_id=pk)
+        messages.success(request, "You activated new challenge!")
+        return HttpResponseRedirect(reverse("challenge_detail", args=[str(pk)]))
+
+
+def event_view_done(request):
+    """gain points when user challenge is completed by clicking >done< button;
+    should check if points can be added (depending on duration and frequency in model)"""
+    try:
+        challenge_id = int(request.POST.get("done"))
+    except ValueError:
+        messages.warning(request, "This challenge does not exist")
+    else:
+        if Challenge.objects.filter(pk=challenge_id).exists():
+            points = Challenge.objects.get(pk=challenge_id).points
+            # check also not active user_challenges and their logs for frequency!
+            if UserChallenge.objects.filter(
+                user_id=request.user, challenge_id=challenge_id
+            ).exists():
+                user_challenge = UserChallenge.objects.get(
+                    user_id=request.user,
+                    challenge_id=challenge_id,
+                    is_active=True,
+                    is_deleted=False,
+                )
+                if user_challenge:
+                    user_challenge.check_if_active
+                    if user_challenge.is_active:
+                        """proceed user getting points for user challenge; it is up to date"""
+                        new_points = user_challenge.get_points
+                        if new_points == points:
+                            Log.objects.create(
+                                user_challenge_id=user_challenge.id, points=points
+                            )
+                            my_profile = request.user.profile
+                            my_profile.points += points
+                            my_profile.save()
+                            messages.success(
+                                request,
+                                f"You have just got {new_points} points for challenge: {user_challenge.challenge}!",
+                            )
+                        else:
+                            messages.warning(
+                                request,
+                                f"You cannot get new points for this challenge yet: {user_challenge.challenge}. "
+                                f"You have already done it within last {user_challenge.challenge.frequency} days",
+                            )
+                    else:
+                        # try to check previous logs, even for other user_challenges, but for the same user and challenge
+                        # and even if they are not active or are deleted, it counts points!
+                        prev_logs = Log.objects.filter(
```

Comment:
możesz spokojnie użyć `user_challenge.logs`

Reply:
no właśnie chyba nie mogę... bo logi mam podpięte do user_challenge, a jak dezaktywuję jeden user_challenge, to dla tego samego usera drugie user_challenge będzie miało już inne pk... a ja chcę sprawdzić właśnie te poprzednie :/

Reply:
Ok, 
Ale i tak nie jestem pewna czy dobrze Ci to działa - filtrujesz `user_challenge__user_id=request.user`

Reply:
wydaje mi się, że filtruję logi dla wszystkich UserChallenge'ów powiązanych z user_id o wartości request.user ;
sprawdzę to sobie jutro bezpośrednio w bazie danych przez shella

Reply:
Przeczytaj jeszcze raz co napisałaś ;)

Reply:
W toku dalszych prac nad kodem zmieniłam sporo w tej funkcji, jednak w tym fragmencie upieram się przy swoim rozwiązaniu z filtrowaniem z `Log.objects`. Sprawdziłam w shellu i to poprawny zapis filrtowania.
Nie chcę logów tylko do ostatniego user_challenge, bo zakładam, że ktoś mógł zatrzymać dwa takie same wyzwania, ale w pierwszym miał logi, a w drugim (późniejszym) już nie, więc user_challenge.log_set.all() dałby mi tylko logi ostatniego i umożliwił zdobycie punktów dla nie przekroczonego frequency. 
Może być tak, że nie zrozumiałam Twojej sugestii odnośnie użycia `user_challenge.logs`, ale zastanawiałam się jak to zastosować, i nic nie działało poprawnie.